### PR TITLE
Add TableId to "getPolicy" method V3

### PR DESCRIPTION
**Pull request by sanderpick**
@awmuncy @joewagner here is yet another approach that simply adds the new method to the existing controller. The benefit here is more "conceptual": we're not versioning anything, just adding new methods. It's a little funky because users will only want one of the methods. There's an abstract contract that we'd recommend people to inherit from that does not allow overriding the old `getPolicy(address)` method. However, everything works fine if a contract was to inherit directly from the interface, it's a just a little awkward. We could add additional methods to the interface as needed. 

I'll probably marinate on this a bit. I am wondering if we just want to kick this change down the road until we figure out the next full iteration of ACL and policies.

Fixes EVM-19

### CODE REVIEW

**Comment by awmuncy** [comment]
For me, it seems like having both method signatures is confusing. I can imagine thinking "When is one used over the other, can you use both?" When a dev looks at the interface, I think it should A) Point you in the right direction quickly B) Reduce real and perceived conceptual overhead. I think this cuts against that. 

Also, I see the appeal of kicking it down the road until we have everything ACL figured out. That said, I think this has been a blocker for several people. You, (since you added the ticket), myself, and Andrew have all run into not being able to pass in TableID as a very real blocker, not just a conceptual nice-to-have. That is to say, I think we should deal with this issue now (I like your V2 approach), and if there are large changes to how the ACL works, changes are going to be tricky even if we don't solve this issue now. 

I can be convinced otherwise on both points, but those are my initial thoughts.

**Comment by sanderpick** [comment]
Fair points. I agree, two methods that do the same thing w/ slightly diff signatures is confusing. 

Okay, let's move back to the V2 approach. Give it another review when you get a chance.

**Comment by joewagner** [comment]
Both this and the 2 approach seem like they work well.  I'm leaning toward a preference for this PR.  

This PR results in slightly lower gas, but probably not enough to warrant any preference.

Concerning standard practices / readability: at first glance the variadic/polymorphic pattern might seem confusing, but I looked for examples of what other projects are doing in this kind of situation, and it seems like a fairly common approach when combined with a `@deprecated` message.  Just cause other's are doing it doesn't mean we should, but after reading SO posts and looking at what's being done by others I don't think this approach would be too confusing.

Here's a couple places I've found projects doing this.
The Arbitrum Sequencer Inbox uses this pattern.
Check out the `addSquencerL2BatchFromOrigin` functions [here](https://etherscan.io/address/0x1c479675ad559dc151f6ec7ed3fbf8cee79582b6#writeProxyContract).
One of them has a deprecated message, which comes from the docstring [here](https://etherscan.io/address/0xd03bfe2ce83632f4e618a97299cc91b1335bb2d9#code#F1#L172), and will _probably_ be rendered in IDEs and contract explorers.
Also, Ethers recently released v6 with features that make interacting with this kind of contract a lot easier. If you specify the types of your function call everything works, details are [here](https://docs.ethers.org/v6/migrating/#migrate-contracts).

----

RE: when to do the upgrade.
My opinion is it seems good to incorporate as many features as we can into a v1/v2 contract and deploy that sometime in season 6 as part of a Ship.

----

**This is more of comment on all the approaches, but leaving it here for lack of a better spot.** 
In case it's helpful, I've been experimenting with these changes with the following setup:
The chess app has a branch that works with the tableland's newest package versions [here](https://github.com/tablelandnetwork/example-chess-nft/tree/joe/protocol-upgrade).
And there's a loose set of commands that I've been using to setup the local env [here](https://gist.github.com/joewagner/5d0266d94363f0e07a282c24703a51f7).

**Comment by sanderpick** [comment]
@joewagner thanks for the good sleuthing. I looked at the various solutions again w/ fresh eyes. I'm leaning toward this one as well:
- Lower gas.
- If we adopt the use of `@notice DEPRECATED` in doc strings as Arbitrum does, this provides a cleaner path for more changes.

I'll clean this up and ask for a final review.

**Comment by sanderpick** [comment]
ok @joewagner @awmuncy this is cleaned up and ready for final review.

File: contracts/TablelandTables.sol [hunk 1]
```diff
@@ -117,7 +148,7 @@ contract TablelandTables is
         }
```
Comment (awmuncy) [inline contracts/TablelandTables.sol:148]: I noticed the conditional statement on line 146 and I was wondering if you could help me understand it better. Specifically, I'm curious about the scenario in which the controller is set to the 0 address. From reading the code, it seems like when the controller is set to the 0 address, all operations are allowed. Can you confirm if this is the intended behavior or if the zero address should have no permissions?
(This comment isn't actually related to changed node, apparently this logic has been the case since before this pull request, just doing a sanity check.)

File: contracts/interfaces/ITablelandController.sol [hunk 2]
```diff
@@ -1,4 +1,29 @@
...
+    /**
+     * @dev Returns a {TablelandPolicy} struct defining how a table can be accessed by `caller`.
+     */
+    function getPolicy(
+        address caller,
+        uint256 tableId
+    ) external payable returns (TablelandPolicy memory);
+
+    /**
+     * @dev Returns a {TablelandPolicy} struct defining how a table can be accessed by `caller`.
```
Comment (awmuncy) [inline contracts/interfaces/ITablelandController.sol:24]: Should this interface have a deprecated noticed as well?

**Review by awmuncy — APPROVED** [review]
Looks good to me.

File: contracts/interfaces/ITablelandController.sol (hunk 2 above)
Comment (sanderpick) [reply contracts/interfaces/ITablelandController.sol]: yep, good catch. just pushed a change.

File: contracts/TablelandTables.sol (hunk 1 above)
Comment (sanderpick) [reply contracts/TablelandTables.sol:148]: See the doc here: https://github.com/tablelandnetwork/evm-tableland/blob/main/contracts/ITablelandTables.sol#L103

tl;dr, controller = 0 is a special case that is known by the validator. It's confusing... which is part of the motivation for moving to a unified contract-based access control: EVM-22

**Review by awmuncy — APPROVED** [review]
LGTM
